### PR TITLE
C++ API: torch::nn::LogSoftmax

### PR DESCRIPTION
**Pull request by nuka137**
Add torch::nn::LogSoftmax module and functional support for the C++ API.

Related Issue: #25883 

Reviewer: @yf225

### CODE REVIEW

File: torch/csrc/api/include/torch/nn/options/activation.h [hunk 1]
```diff
@@ -56,5 +57,20 @@ struct LeakyReLUOptions {
...
 };
 
+// ============================================================================
+
+/// Options for LogSoftmax functional and module.
+struct TORCH_API LogSoftmaxOptions {
+  LogSoftmaxOptions(int dim = -1, torch::Dtype dtype = torch::Dtype::Undefined);
+
+  // Dimension along which LogSoftmax will be computed.
+  TORCH_ARG(int, dim);
```
Comment (pbelevich) [inline torch/csrc/api/include/torch/nn/options/activation.h]: Please change it to c10::optional<int64_t> with c10::nullopt default value

**Review by pbelevich — CHANGES_REQUESTED** [review]

**Comment by yf225** [comment]
Thanks a lot for the great work! We will need some fixes as suggested in https://github.com/pytorch/pytorch/pull/27446.

**Comment by nuka137** [comment]
@yf225

Fixed all issues discussed in #27446, and rebased to newest master branch.
Could you review this PR again?

**Review by yf225 — APPROVED** [review]
Thanks so much for the great work @nuka137!

**Pull request closed without merging**